app/konnect/org-management/audit-logging/verify-signatures.md

```diff
@@ -0,0 +1,139 @@
+    ```
+
+    The response should look something like this:
+
+    ```json
+    {
+        "keys": [
+            {
+                "alg": "EdDSA",
+                "crv": "Ed25519",
+                "kid": "1d4608c22e448672d5386b4071b70442as45c58265",
+                "kty": "OKP",
+                "x": "aFNAu9QEQhiunrGuyS14ePHzoOb2vash783p1-_Nrc3M"
+            }
+        ]
+    }
+    ```
+1. Extract the signature of the entry to calculate the signature to match. 
+
+1. Decode the Base64 URL-encoded signature into bytes.
+
+1. Remove the signature value from the entry.
+
+   The adjusted entry will look slightly different depending on the format that you're using. 
+   See the [CEF verification payload](#cef-verification-payload) and 
+   [JSON verification payload](#json-verification-payload) sections for examples.
+
+1. Verify the ED25519 signature with the public key, signatureless audit log entry, and decoded signature.
+
+## CEF verification payload
+
+For a CEF entry, the signature is computed on the CEF value with the signature field, value, and 
+preceding whitespace removed. 
+
+For example, this CEF entry:
+
+```
+Apr 14 05:39:08 
+konghq.com 
+CEF:0|KongInc|Konnect|1.0|konnect|Authz.usage|1|rt=1681450748406 
+src=127.0.0.6 
+action=retrieve 
+granted=true 
+org_id=b065b594-6afc-4658-9101-5d9cf3f36b7b 
+principal_id=87655c36-8d63-48fe-9a1e-53b28dfbc19b 
+trace_id=3895213347334635099 
+user_agent=grpc-node/1.24.11 
+grpc-c/8.0.0 (linux; chttp2; ganges) 
+sig=fnjOUISTC2uLWTqCcmwUuOWa_daa4vEPxxUn2-QPVU1dmB-Ed2y2mwoB9dq94JJz2tF6Ok41Bci4FGwk4FhxAg
+```
+
+Would have the signature calculated on this payload:
+
+```
+Apr 14 05:39:08 
+konghq.com 
+CEF:0|KongInc|Konnect|1.0|konnect|Authz.usage|1|rt=1681450748406 
+src=127.0.0.6 
+action=retrieve 
+granted=true 
+org_id=b065b594-6afc-4658-9101-5d9cf3f36b7b 
+principal_id=87655c36-8d63-48fe-9a1e-53b28dfbc19b 
+trace_id=3895213347334635099 
+user_agent=grpc-node/1.24.11 
+grpc-c/8.0.0 (linux; chttp2; ganges)
+```
+
+## JSON verification payload
+
+For a JSON entry, the `sig` field, value, and prepended comma is removed from the entry. 
+
+For example, this JSON entry:
+
+```json
+{
+    "action": "read",
+    "cef_version": "0",
+    "event_class_id": "identity",
+    "event_product": "Konnect",
+    "event_ts": "2023-04-28T20:52:09Z",
+    "event_vendor": "KongInc",
+    "event_version": "1.0",
+    "granted": true,
+    "name": "Authz.identity-provider",
+    "org_id": "b065b594-6afc-4658-9101-5d9cf3f36b7b ",
+    "principal_id": "87655c36-8d63-48fe-9a1e-53b28dfbc19b ",
+    "rt": 1682715129807,
+    "severity": 1,
+    "src": "127.0.0.6",
+    "trace_id": 3895213347334635099,
+    "user_agent": "grpc-go/1.54.0",
+    "sig": "Jm73seSwAiacSBysDmgQ3D_R_1c39_T0Iuus9GqUEnAjCc-UnyKhFgS8jDx5UIl4J6PTCH5ouuPizwnjnMDECg"
+}
+```
+
+Would have the signature calculated on this JSON payload:
+
+{:.note}
+> **Note**: Don't add any extra whitespace or make any formatting changes to the JSON entries.
```

Comment:
Due to this requirement, I think we cannot "pretty print" any of the JSON or CEF examples.

The message they would have to calculate the signature with literally has to be something like:
```JSON
{"action":"update","cef_version":"0","event_class_id":"auditlogs","event_product":"Konnect","event_ts":"2023-06-07T23:59:26Z","event_vendor":"KongInc","event_version":"1.0","granted":true,"name":"Authz.webhook","org_id":"2e2eca38-cf13-487f-90a7-2ab8b6b1fd77","principal_id":"33f7eed7-a407-4b0d-9b06-221afc45371f","rt":1686182366202,"severity":1,"src":"127.0.0.6","trace_id":5733225434988743696,"user_agent":"grpc-go/1.55.0"}
```
or
```CEF
2023-06-08T18:56:02Z konghq.com CEF:0|KongInc|Konnect|1.0|AUTHENTICATION_TYPE_PAT|AUTHENTICATION_OUTCOME_SUCCESS|0|rt=1686250562822 src=127.0.0.1 request=/api/v1/personal-access-tokens/introspect success=true org_id=2e2eca38-cf13-487f-90a7-2ab8b6b1fd77 principal_id=33f7eed7-a407-4b0d-9b06-221afc45371f trace_id=5121050581507016718 user_agent=grpc-go/1.55.0
```